lib/matobj.gi

```diff
@@ -1568,3 +1568,572 @@ InstallMethod( \[\,\]\:\=, "for a matrix object, two positions, and an object",
+  [ IsMatrixObj and IsMutable, IsInt, IsObject ],
+  function( mat, row, scalar )
+    local i;
+    
+    # Checks
+    if not( 0 < row and row <= NrRows(mat) ) then
+        Print("The second argument row has to fulfill 0 < row <= NrRows(mat) ");
+        return fail;
+    fi;
+  
+    for i in [1..NrCols(mat)] do
+        mat[row,i] := scalar * mat[row,i];
+    od;
+    
+  end );
+
+
+# TODO FOR MUTABLE MATRICES
+# InstallMethod( MultMatrixRow, "for a mutable matrix object, one row number, and an scalar",
+#  [ IsMatrixObj, IsInt, IsObject ],
+# function( mat, row, scalar )
+#    local One, MatRow, i, CopyMat;
+    
+    # Checks
+#    if not( 0 < row and row < NrRows(mat) ) then
+#        Print("The second argument row has to fullfill 0 < row < NrRows(mat) ");
+#        return fail;
+#    fi;
+  
+#    CopyMat := ShallowCopy(mat);
+  
+#    for i in [1..NrCols(mat)] do
+#        CopyMat[row,i] := scalar * CopyMat[row,i];
+#    od;
+    
+    # return Matrix(CopyMat,mat);
+    
+#  end );
+
+
+############################################################################
+##
+#O  MultMatrixColumn( <M>, <row>, <scalar> )
+##
+##  <#GAPDoc Label="MultMatrixColumn">
+##  <ManSection>
+##  <Oper Name="MultMatrixColumn" Arg='mat,i,elm' Label=""/>
+##  
+##  <Returns>nothing</Returns>
+##
+##  <Description>
+##  <P/>
+##  For a matrix <A>mat</A> such that <A>mat</A> is mutable, a column index <A>i</A>  
+##  and a scalar <A>elm</A> this operation multiplies the <A>i</A>-th column with <A>elm</A> 
+##  in-place.
+##  Note that by default <A>elm</A> is multiplied from the right. 
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+InstallMethod( MultMatrixColumn, "for a mutable matrix object, one column number, and an scalar",
+  [ IsMatrixObj and IsMutable, IsInt, IsObject ],
+  function( mat, column, scalar )
+    local i;
+    
+    # Checks
+    if not( 0 < column and column <= NrCols(mat) ) then
+        Print("The second argument row has to fulfill 0 < row <= NrCols(mat) ");
+        return fail;
+    fi;
+  
+    for i in [1..NrRows(mat)] do
+        mat[i,column] := mat[i,column] * scalar;
+    od;
+    
+  end );
+  
+  
+
+############################################################################
+##
+#O  MultMatrixRowLeft( <M>, <row>, <scalar> )
+##
+##  <#GAPDoc Label="MultMatrixRowLeft">
+##  <ManSection>
+##  <Oper Name="MultMatrixRowLeft" Arg='mat,i,elm' Label=""/>
+##  
+##  <Returns>nothing</Returns>
+##
+##  <Description>
+##  <P/>
+##  For a matrix <A>mat</A> such that <A>mat</A> is mutable, a row index <A>i</A>  
+##  and a scalar <A>elm</A> this operation multiplies <A>elm</A> with the <A>i</A>-th row 
+##  in-place.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+InstallMethod( MultMatrixRowLeft, "for a mutable matrix object, one row number, and an scalar",
+  [ IsMatrixObj and IsMutable, IsInt, IsObject ],
+  function( mat, row, scalar )
+ 
+    MultMatrixRow(mat,row,scalar);
```

Comment:
I would put the explicit code into a method of `MultMatrixRowLeft`, and call `MultMatrixRowLeft` in the method for `MultMatrixRow`; then calling the more specific operation directly would save one indirection.